### PR TITLE
feat(meru templates): data model

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

First step towards https://github.com/dust-tt/dust/issues/4373

Will likely evolve in the future. This first draft is based on examples from here: https://www.notion.so/dust-tt/Templates-33c161f77a7d4015b1fae83e29a391a8.

Each template has its own name and optionally a description (which will be displayed in the new "Create Assistant" modal that comes right before the builder). 
There are fields to store the presets for the action type, the instructions, the model, the temperature, the handle and the description.
There are fields to store the "help" bubble for the instructions and actions steps of the builder. This will maybe change -- maybe we want several bubbles at each step. Maybe we'll want info bubbles at sharing and naming steps. Doesn't matter yet.

There's also another table to store the tags.

## Deploy Plan

requires migrating the DB.

### CODE REVIEW

File: front/lib/models/templates.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare sId: string;
+  declare name: string;
+  declare description: string | null;
+
+  declare visibility: "draft" | "published" | "disabled";
+
+  declare presetHandler: string | null;
```
Comment (flvndvd) [inline front/lib/models/templates.ts]: `presetHandle`?

**Review by spolu — APPROVED** [review]
LGTM

Resource pattern from the get go please :pray: :)

**Review by flvndvd — APPROVED** [review]

File: front/lib/models/templates.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: frontSequelize,
+    modelName: "templateTag",
+    indexes: [
+      { fields: ["templateId", "tag"], unique: true },
+      { fields: ["tag"] },
```
Comment (flvndvd) [inline front/lib/models/templates.ts]:
```suggestion
      { fields: ["tag", "templateId"], unique: true },
```

Comment (fontanierh) [reply front/lib/models/templates.ts]: :check

File: front/lib/models/templates.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/models/templates.ts]: :check

File: front/lib/models/templates.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+export class TemplateTag extends Model<
+  InferAttributes<TemplateTag>,
+  InferCreationAttributes<TemplateTag>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare templateId: ForeignKey<Template>;
+  declare tag: string;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/models/templates.ts:124]: yep

File: front/lib/models/templates.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  declare presetHandle: string | null;
+  declare presetDescription: string | null;
+  declare presetInstructions: string | null;
+  declare presetTemperature:
+    | "deterministic"
+    | "factual"
+    | "balanced"
+    | "creative"
+    | null;
+  declare presetModelId: string | null;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/models/templates.ts:34]: Thanks

**Comment by fontanierh** [comment]
resource patterned @spolu @flvndvd

**Review by flvndvd — COMMENTED** [review]
LGTM

**Pull request merged**